client/src/views/Environment.tsx

```diff
@@ -34,36 +40,55 @@ function Environment() {
     setEndpoints(list);
   }, ENDPOINTS_POLLING_INTERVAL);
 
+
+    useInterval( async () => {
+      if (!environment || !currentContext) return;
```

Comment:
Can we group this logic into a `refreshStatus` function, that helps understand what the logic is for, or at least we can add a comment about it at the beginning of the arrow function (see `refreshContext` as an example)